superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndFilterSelect.test.tsx

```diff
@@ -153,9 +170,187 @@ test('renders options with adhoc metric', async () => {
     }),
     {
       useDnd: true,
+      store,
     },
   );
   expect(
     await screen.findByText('Drop columns/metrics here or click'),
   ).toBeInTheDocument();
 });
+
+test('cannot drop a column that is not part of the simple column selection', () => {
```

Comment:
@kgabryje I also added the spec for your case. Please let me know if you have any issues.